File: macos/QMK Toolbox/USB/Bootloader/AVRISPDevice.m
#import "AVRISPDevice.h"

@implementation AVRISPDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"AVR ISP";
        self.type = BootloaderTypeAVRISP;
        while (self.serialPort == nil) {
            self.serialPort = [self.usbDevice findSerialPort];
        }
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"avrisp", @"-U", [NSString stringWithFormat:@"flash:w:%@:i", file], @"-P", self.serialPort, @"-C", @"avrdude.conf"]];
}

-(void)flashEEPROMWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"avrisp", @"-U", [NSString stringWithFormat:@"eeprom:w:%@:i", file], @"-P", self.serialPort, @"-C", @"avrdude.conf"]];
}

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ [%@]", [super description], self.serialPort];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/AVRISPDevice.swift
import Foundation

class AVRISPDevice: BootloaderDevice {
    private var serialPort: String?

    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "AVR ISP"
        type = .avrIsp
        eepromFlashable = true
        while serialPort == nil {
            serialPort = findSerialPort()
        }
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "avrisp", "-U", "flash:w:\(file):i", "-P", serialPort!, "-C", "avrdude.conf"])
    }

    override func flashEEPROM(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "avrisp", "-U", "eeprom:w:\(file):i", "-P", serialPort!, "-C", "avrdude.conf"])
    }

    override var description: String {
        "\(super.description) [\(serialPort!)]"
    }
}
